docs/blueprints/deletion-performance.md

```diff
@@ -0,0 +1,39 @@
+# Deletion Performance
+
+## Abstract
+
+This document describes design for improvments to deletion speed.
+
+## Background
+
+Current design requires uplinks to send deletion information to all the storage nodes. This ends up taking a considerable amount of time.
+
+There are few goals with regards to deletions:
+
+- We need to ensure that uplinks are responsive.
+- We need to ensure that storage nodes aren't storing too much garbage, since it reduces overall capacity of the network.
+- We need to ensure that satellites aren't storing too much garbage, since that increases running cost of the satellite.

```

Comment:
How satellite could store garbage in a delete?
I understand that the garbage data in the storage node side, but I don't get it in the satellite.

Reply:
For example, we could use tombstones on the pointers, marking them as deleted and then sending delete requests as a batch to the satellite. Similarly we could keep a list of pieces to delete per storage node. Effectively, this would mean that satellite is holding on to garbage so that it could be collected later.